packages/jest-message-util/src/index.ts

```diff
@@ -5,323 +5,296 @@
-    return line;
-  }
-
-  let filePath = slash(path.relative(config.rootDir, match[2]));
-  // highlight paths from the current test file
-  if (
-    (config.testMatch &&
-      config.testMatch.length &&
-      micromatch([filePath], config.testMatch).length > 0) ||
-    filePath === relativeTestPath
-  ) {
-    filePath = chalk.reset.cyan(filePath);
-  }
-  return STACK_TRACE_COLOR(match[1]) + filePath + STACK_TRACE_COLOR(match[3]);
+  options: StackTraceOptions = {},
+  relativeTestPath: Path | null = null,
+): JestVisitorOptions => {
+  const {rootDir, testMatch} = config;
+  const {
+    noStackTrace = false,
+    noCodeFrame = false,
+    indentHeader = '    ',
+    indentFrames = '    ',
+  } = options;
+  return {
+    frames: !noStackTrace,
+    indentFrames,
+    indentHeader,
+    noCodeFrame: noStackTrace || noCodeFrame,
+    noStackTrace,
+    relativeTestPath,
+    rootDir,
+    testMatch,
+  };
 };
 
 export const getStackTraceLines = (
-  stack: string,
+  frames: string,
   options: StackTraceOptions = {noCodeFrame: false, noStackTrace: false},
-): Array<string> => removeInternalStackEntries(stack.split(/\n/), options);
+): Array<string> => {
+  const parsedError = parseError(frames);
 
-export const getTopFrame = (lines: Array<string>): Frame | null => {
-  for (const line of lines) {
-    if (line.includes(PATH_NODE_MODULES) || line.includes(PATH_JEST_PACKAGES)) {
-      continue;
-    }
+  parsedError.frames = cleanFrames(parsedError.frames, options);
 
-    const parsedFrame = stackUtils.parseLine(line.trim());
+  return parsedError.frames
+    ? parsedError.frames.map(frame => printNode(frame)!)
+    : [];
+};
 
-    if (parsedFrame && parsedFrame.file) {
-      return parsedFrame as Frame;
-    }
-  }
+export const formatStackTrace = (
+  error: string | ErrorNode,
+  config: StackTraceConfig,
+  options?: StackTraceOptions,
+  testPath?: Path,
+): string => {
+  const visitorOptions = {
+    ...buildVisitorOptions(config, options, testPath),
+    frames: true,
+    indentFrames: '    ',
+    indentHeader: '',
+  };
+  const parsedError = typeof error === 'string' ? parseError(error) : error;
 
-  return null;
+  parsedError.frames = cleanFrames(parsedError.frames, options);
+
+  return '\n' + JestVisitor.visit(parsedError, visitorOptions) || '';
 };
 
-export const formatStackTrace = (
-  stack: string,
+export const getTopFrame = (frames: Array<string>): Frame | null => {
+  const frameNode = __getTopFrame(frames.map(frame => parseFrame(frame)));
+  const site = frameNode?.callSite?.site || null;
+
+  return (
+    site && {
+      column: site.position.column,
+      file: site.locator.path,
+      line: site.position.line,
+    }
+  );
+};
+
+// ExecError is an error thrown outside of the test suite (not inside an `it` or
+// `before/after each` hooks). If it's thrown, none of the tests in the file
+// are executed.
+export const formatExecError = (
+  error: Error | TestResult.SerializableError | string | undefined,
```

Comment:
It might make sense to have `error: unknown` here. In a way `number`, `null`, or some object would be handled. Type could allow these. I was wrestling with this idea for some time till my head remembered this PR. Just leaving a note.